### ISSUE TITLE
Can't fully flash alpha CHIP

### ISSUE THREAD

**Issue by mrlowalowa** [NextThingCo/CHIP-SDK#16]
The mainproblem is this while flashing from vagrant. All gits are uptodate..

'libusb usb_bulk_send error -7'

Full output:

> vagrant@vagrant-ubuntu-trusty-32:~/CHIP-tools$ ./chip-update-firmware.sh -d
waiting for fel...OK
debian selected
ROOTFS_URL=http://opensource.nextthing.co.s3.amazonaws.com/chip/debian/stable/46
BUILD=46
BR_URL=http://opensource.nextthing.co/chip/buildroot/stable/73/images
BR_BUILD=73
/home/vagrant/CHIP-tools/.firmware/images/rootfs.ubi exists... comparing to http://opensource.nextthing.co.s3.amazonaws.com/chip/debian/stable/46/rootfs.ubi
MD5: a121f4654dec1c3d02ea863223d8a2b0
S3_MD5: a121f4654dec1c3d02ea863223d8a2b0
file already downloaded
/home/vagrant/CHIP-tools/.firmware/images/sun5i-r8-chip.dtb exists... comparing to http://opensource.nextthing.co/chip/buildroot/stable/73/images/sun5i-r8-chip.dtb
MD5: 09707eb7aa4709365cc14aa0314e4716
S3_MD5: 09707eb7aa4709365cc14aa0314e4716
file already downloaded
/home/vagrant/CHIP-tools/.firmware/images/sunxi-spl.bin exists... comparing to http://opensource.nextthing.co/chip/buildroot/stable/73/images/sunxi-spl.bin
MD5: 1dd925fa15a21bc60dd2172fe3ac40ad
S3_MD5: 1dd925fa15a21bc60dd2172fe3ac40ad
file already downloaded
/home/vagrant/CHIP-tools/.firmware/images/sunxi-spl-with-ecc.bin exists... comparing to http://opensource.nextthing.co/chip/buildroot/stable/73/images/sunxi-spl-with-ecc.bin
MD5: f96fc379e59673e23b3fc1b99a89f569
S3_MD5: f96fc379e59673e23b3fc1b99a89f569
file already downloaded
/home/vagrant/CHIP-tools/.firmware/images/uboot-env.bin exists... comparing to http://opensource.nextthing.co/chip/buildroot/stable/73/images/uboot-env.bin
MD5: 6f2b79a781f9f490911012ec3aa653e9
S3_MD5: 6f2b79a781f9f490911012ec3aa653e9
file already downloaded
/home/vagrant/CHIP-tools/.firmware/images/zImage exists... comparing to http://opensource.nextthing.co/chip/buildroot/stable/73/images/zImage
MD5: 91614d8d67bd87987ea5bc958e61cfc3
S3_MD5: 91614d8d67bd87987ea5bc958e61cfc3
file already downloaded
/home/vagrant/CHIP-tools/.firmware/images/u-boot-dtb.bin exists... comparing to http://opensource.nextthing.co/chip/buildroot/stable/73/images/u-boot-dtb.bin
MD5: a9051ff21d462ecc66d7ee9ded2ebe0a
S3_MD5: a9051ff21d462ecc66d7ee9ded2ebe0a
file already downloaded
BUILDROOT_OUTPUT_DIR = /home/vagrant/CHIP-tools/.firmware
== preparing images ==
PADDED_SPL_SIZE=0x000000c4
0+1 records in
1+0 records out
4194304 bytes (4.2 MB) copied, 0.00378882 s, 1.1 GB/s
UBOOT_SIZE=0x00400000
PADDED_UBOOT_SIZE=0x400000
0+0 records in
0+0 records out
0 bytes (0 B) copied, 0.00956901 s, 0.0 kB/s
echo Configuring Video Mode
reset
Image Name:   flash CHIP
Created:      Thu Mar 10 23:01:39 2016
Image Type:   ARM Linux Script (uncompressed)
Data Size:    1022 Bytes = 1.00 kB = 0.00 MB
Load Address: 00000000
Entry Point:  00000000
Contents:
   Image 0: 1014 Bytes = 0.99 kB = 0.00 MB
== upload the SPL to SRAM and execute it ==
waiting for fel...OK
libusb usb_bulk_send error -7
ERROR: could not flash
vagrant@vagrant-ubuntu-trusty-32:~/CHIP-tools$ fel
Usage: fel [options] command arguments... [command...]
	-v, --verbose			Verbose logging

	spl file			Load and execute U-Boot SPL
		If file additionally contains a main U-Boot binary
		(u-boot-sunxi-with-spl.bin), this command also transfers that
		to memory (default address from image), but won't execute it.

	uboot file-with-spl		like "spl", but actually starts U-Boot
		U-Boot execution will take place when the fel utility exits.
		This allows combining "uboot" with further "write" commands
		(to transfer other files needed for the boot).

	hex[dump] address length	Dumps memory region in hex
	dump address length		Binary memory dump
	exe[cute] address		Call function address
	read address length file	Write memory contents into file
	write address file		Store file contents into memory
	ver[sion]			Show BROM version
	clear address length		Clear memory
	fill address length value	Fill memory
vagrant@vagrant-ubuntu-trusty-32:~/CHIP-tools$ fel -v
vagrant@vagrant-ubuntu-trusty-32:~/CHIP-tools$ ./chip-update-firmware.sh -d
waiting for fel.....^C^C
vagrant@vagrant-ubuntu-trusty-32:~/CHIP-tools$ fel -v
vagrant@vagrant-ubuntu-trusty-32:~/CHIP-tools$ ./chip-update-firmware.sh -d
waiting for fel........

AND after several tries I got this one..

./chip-fel-flash.sh: line 195: img2simg: command not found

**Comment by mrlowalowa** [comment]
So the libusb seems to be due to the fact that the image is to big for libusb. fastflash uses 30mb packages.

Img2simg fixed by installing adroid utilities. (>img2simg ---> try to install androindXY-->su PW:vagrant)

But now I get this:

>    Image 0: 1004 Bytes = 0.98 kB = 0.00 MB
== upload the SPL to SRAM and execute it ==
waiting for fel...OK
== upload spl ==
== upload u-boot ==
== upload u-boot script ==
== execute the main u-boot binary ==
== creating sparse image ==
== waiting for fastboot ==
waiting for fastboot...................OK
< waiting for device >

But I'm waiting since 1/4 hour already.. (did this twice)

**Comment by jasonabele** [comment]
Regarding the waiting for fastboot, I have had a similar issue, and the root cause was that the usb layer did not initialize within uboot.  You can check for this by watching the console output on UART1, I forget the exact string and do not have the specific failed CHIP on my desk anymore.  It should be something like "No devices found" in the uboot output.

**Comment by mrlowalowa** [comment]
my console output is:

musb-hdrc: peripheral reset irq lost!

and I'm continuing to wait. When I abort it with ^C over the UART it says flash finished but can't mount the storage and cannot boot...

**Comment by mrlowalowa** [comment]
It's working now.

I managed it by updating the firmware without fast boot (which I thought was not possible due to the size issue of debian and the normal mode, fast boot was using smaller packages of 30 mb or so)

Another thing I did different was that I wasn't replugging or restarting the whole system at all. I just had been a little bit lazy and figured out how to 'restart' the CHIP from the emergency uboot that was accessible right there.. After cross reading all options the help gave me I choose `reset` as the suitable script. I used this like 3-4 times with fast boot and no success and the last time without the -f and it took like 10 min but it worked!
